src/libcrun/container.c

```diff
@@ -2016,12 +2016,80 @@ cleanup_watch (libcrun_context_t *context, runtime_spec_schema_config_schema *de
 }
 
 static int
-open_seccomp_output (const char *id, int *fd, bool readonly, const char *state_root, libcrun_error_t *err)
+store_seccomp_cache (const char *state_root, const char *id, char checksum[65], libcrun_error_t *err)
+{
+  cleanup_free char *seccomp_cache_dir = NULL;
+  cleanup_free char *seccomp_cache = NULL;
+  cleanup_free char *src_path = NULL;
+  cleanup_free char *dir = NULL;
+  struct stat st;
+  int ret;
+
+  dir = libcrun_get_state_directory (state_root, id);
+  if (UNLIKELY (dir == NULL))
+    return crun_make_error (err, 0, "cannot get state directory");
+
+  ret = append_paths (&src_path, err, dir, "seccomp.bpf", NULL);
+  if (UNLIKELY (ret < 0))
+    return ret;
+
+  ret = append_paths (&seccomp_cache_dir, err, dir, "..", ".seccomp-cache", NULL);
+  if (UNLIKELY (ret < 0))
+    return ret;
+
+  ret = stat (seccomp_cache_dir, &st);
+  if (ret == 0 && st.st_size > 500)
```

Comment:
Small question: I am unable to find context for magic size check for `500` bytes , is it something generic ?

Reply:
no it is just a magic number to avoid that the cache grows indefinitely, I'll have to clean this up